src/skeleton/mod.rs

```diff
@@ -29,9 +29,15 @@ pub(in crate::skeleton) struct ParsedManifest {
 
 impl Skeleton {
     /// Find all Cargo.toml files in `base_path` by traversing sub-directories recursively.
-    pub fn derive<P: AsRef<Path>>(base_path: P) -> Result<Self, anyhow::Error> {
+    pub fn derive<P: AsRef<Path>>(
+        base_path: P,
+        member: Option<String>,
+    ) -> Result<Self, anyhow::Error> {
         // Read relevant files from the filesystem
         let config_file = read::config(&base_path)?;
+        if let Some(member) = member {
+            replace_members_with_specific(&base_path, member)?;
```

Comment:
This makes `derive` produce a side effect (modifies the file system). We do this shortly after `derive` (in `main` after preparing the skeleton), but this seemed easier to me than trying to exclude the top-level Cargo.toml if it's both a workspace and `member` is specified.

Reply:
I am not a fan - we should manipulate the manifest in memory in `derive` and keep it side-effect free.
You should be able to do this easily by iterating over the `serialised_manifests` collection.

Reply:
That makes sense to me! I'm doing it in `manifests`, though, instead of `serialized_manifests`, to avoid serializing then deserializing.